network-api/networkapi/wagtailcustomization/redirects/middleware.py

```diff
@@ -11,7 +11,8 @@ def _new_get_redirect(request, path):
         locale_prefix = f'/{request.LANGUAGE_CODE}/'
         if path.startswith(locale_prefix):
             path = path.replace(locale_prefix, '/', 1)
-
+    if request.META['QUERY_STRING']:
```

Comment:
checking if there is a query string in the request, if so, append it to the updated path so it does not get removed